src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -211,6 +225,12 @@ private SslStreamCertificateContext(X509Certificate2 target, ReadOnlyCollection<
                 return null;
             }
 
+            IntPtr[] issuerHandles = ArrayPool<IntPtr>.Shared.Rent(_privateIntermediateCertificates.Length);
```

Comment:
I'm wondering if we can simply stackalloc the `Span` The chain is very unlikely to be very long. Since we need it only for the call it should be fine IMHO.

Reply:
The method is async, we can't stackalloc

Reply:
I missed that. We could split the method and I'm not sure if it is worth it. Possibly pass in `_privateIntermediateCertificates` and fix it up in the interop.

Reply:
That could be done.  But it would require doing the object to handle conversion once per attempted endpoint each refresh vs just once per refresh (generally there's only one, but there could be multiple and the code deals with that), as the await is inside the loop.